File: src/main/kotlin/com/itmo/microservices/demo/payment/impl/repository/UserAccountFinancialLogRecordRepository.java
package com.itmo.microservices.demo.payment.impl.repository;

import com.itmo.microservices.demo.payment.impl.model.UserAccountFinancialLogRecord;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.UUID;

@Repository
public interface UserAccountFinancialLogRecordRepository extends JpaRepository<UserAccountFinancialLogRecord, UUID> {

    List<UserAccountFinancialLogRecord> findAllByUserId(UUID userId);
}

File: src/main/kotlin/com/itmo/microservices/demo/payment/impl/repository/UserAccountFinancialLogRecordRepository.kt
package com.itmo.microservices.demo.payment.impl.repository

import com.itmo.microservices.demo.payment.api.model.UserAccountFinancialLogRecordDto
import com.itmo.microservices.demo.payment.impl.entity.UserAccountFinancialLogRecordEntity
import org.springframework.data.jpa.repository.JpaRepository
import java.util.UUID

interface UserAccountFinancialLogRecordRepository: JpaRepository<UserAccountFinancialLogRecordEntity, Long> {
    fun getUserAccountFinancialLogRecordEntitiesByOrderId(orderId: UUID): List<UserAccountFinancialLogRecordDto>
}